ocs_ci/helpers/sanity_helpers.py

```diff
@@ -183,3 +182,156 @@ def __init__(self):
+
+    def __init__(self):
+        """
+        Init the sanity managed service class
+        """
+        super(Sanity, self).__init__()
+        # A dictionary of a consumer index per the fio_scale object
+        self.consumer_i_per_fio_scale = {}
+        # The 'create resources on MS consumers' factory. Will be initialized with the
+        # 'create_scale_pods_and_pvcs_using_kube_job_on_ms_consumers' factory using the method
+        # 'init_create_resources_on_ms_factory'
+        self.create_resources_on_ms_consumers_factory = None
+        # All the variables below will be initialized in the method 'init_create_resources_on_ms_factory'.
+        self.scale_count = None
+        self.pvc_per_pod_count = None
+        self.start_io = None
+        self.io_runtime = None
+        self.pvc_size = None
+        self.max_pvc_size = None
+        self.consumer_indexes = None
+
+    def init_create_resources_on_ms_factory(
+        self,
+        create_scale_pods_and_pvcs_using_kube_job_on_ms_consumers,
+        scale_count=None,
+        pvc_per_pod_count=5,
+        start_io=True,
+        io_runtime=None,
+        pvc_size=None,
+        max_pvc_size=30,
+        consumer_indexes=None,
+    ):
+        """
+        Init the 'create resources on MS consumers' factory.
+        This function uses the factory 'create_scale_pods_and_pvcs_using_kube_job_on_ms_consumers'.
+        Note: You need to use this method before using the method 'create_resources_on_ms_consumers'
+
+        Args:
+           create_scale_pods_and_pvcs_using_kube_job_on_ms_consumers (function): Factory for creating scale
+               pods and PVCs using k8s on MS consumers fixture.
+           scale_count (int): No of PVCs to be Scaled. Should be one of the values in the dict
+               "constants.SCALE_PVC_ROUND_UP_VALUE".
+           pvc_per_pod_count (int): Number of PVCs to be attached to single POD
+               Example, If 20 then 20 PVCs will be attached to single POD
+           start_io (bool): Binary value to start IO default it's True
+           io_runtime (seconds): Runtime in Seconds to continue IO
+           pvc_size (int): Size of PVC to be created
+           max_pvc_size (int): The max size of the pvc
+           consumer_indexes (list): The list of the consumer indexes to create scale pods and PVCs.
+               If not specified - if it's a consumer cluster, it creates scale pods and PVCs only
+               on the current consumer. And if it's a provider it creates scale pods and PVCs on
+               all the consumers.
+        """
+        if consumer_indexes:
+            self.consumer_indexes = consumer_indexes
+        elif is_ms_consumer_cluster():
+            self.consumer_indexes = [config.cur_index]
+        else:
+            self.consumer_indexes = config.get_consumer_indexes_list()
+
+        self.create_resources_on_ms_consumers_factory = (
+            create_scale_pods_and_pvcs_using_kube_job_on_ms_consumers
+        )
```

Comment:
why not including the initialization of create_resources_on_ms_factory withing the `__init__()` method? that way, you will always have create_resources_on_ms_consumers_factory set and ready

Reply:
I thought it would be more flexible to do it in another function.
For example, if we want to run two functions with different parameters. 
From my side, we can change it to the `init` method.

Reply:
In that case, you need first to init the `create_resources_on_ms_consumers_factory`
in the method 'init_create_resources_on_ms_factory'.

Reply:
I changed the initialization of `create_resources_on_ms_factory` to be in the `__init__` method